src/csm_testing/lib/iuf_classes.py

```diff
@@ -0,0 +1,257 @@
+        self.resource = os.getenv("IUF_API_URL_RESOURCE", resource)
+
+    def request(self, method, path, payload=None, timeout=None):
+        method = method.upper()
+        assert method in ['GET', 'HEAD', 'DELETE', 'POST', 'PUT',
+                          'PATCH', 'OPTIONS']
+
+        url=self.apiurl + self.resource + path
+
+        headers = dict()
+        try:
+            token = None
+            headers["Authorization"] = f"Bearer {token}"
+        except:
+            if "gw-service" in self.apiurl:
+                raise
+            else:
+                # if we're not using the "official" api and don't get a token just try without it.  Mostly for local testing.
+                pass
+
+        method_func = method.lower()
+        try:
+            if payload:
+                result = getattr(requests, method_func)(url, headers=headers, json=payload, verify=False, timeout=timeout)
+            else:
+                result = getattr(requests, method_func)(url, headers=headers, verify=False, timeout=timeout)
+        except:
+            raise
+
+        # throw an exception for bad status codes
+        result.raise_for_status()
+
+        return result
+
+    def get_stages(self):
+        api_path = f"/stages"
+        try:
+            api_response = self.request("GET", api_path)
+            return api_response
+        except:
+            raise
+
+class ApiInterface(object):
+    def __init__(self, apiurl="https://api-gw-service-nmn.local/apis", resource="/iuf/v1"):
+        self.auth = Auth()
+        self.apiurl = os.getenv("IUF_API_URL", apiurl)
+        self.resource = os.getenv("IUF_API_URL_RESOURCE", resource)
+
+    def request(self, method, path, payload=None, timeout=None):
+        method = method.upper()
+        assert method in ['GET', 'HEAD', 'DELETE', 'POST', 'PUT',
+                          'PATCH', 'OPTIONS']
+
+        url=self.apiurl + self.resource + path
+
+        headers = dict()
+        try:
+            token = self.auth.token
+            headers["Authorization"] = f"Bearer {token}"
+        except:
+            if "gw-service" in self.apiurl:
+                raise
+            else:
+                # if we're not using the "official" api and don't get a token just try without it.  Mostly for local testing.
+                pass
+
+        method_func = method.lower()
+        try:
+            if payload:
+                result = getattr(requests, method_func)(url, headers=headers, json=payload, verify=False, timeout=timeout)
+            else:
+                result = getattr(requests, method_func)(url, headers=headers, verify=False, timeout=timeout)
+        except:
+            raise
+
+        # throw an exception for bad status codes
+        result.raise_for_status()
+
+        return result
+    
+    def activity_exists(self, activity):
+        try:
+            self.get_activity(activity)
+            return True
+        except:
+            return False
```

Comment:
Do we really not want to do any kind of logging about the exception that was raised? This acts the same way if the request was a "404 not found" as it would if it was a "500 service unavailable", or if there was some other non-request-related Python exception that got raised. It seems like this function should return False only in cases where the activity actually doesn't exist. Wouldn't it make more sense for it to raise the exception in the case that some unexpected error happened?

Reply:
This comment from the previous review has not been addressed or responded to

Reply:
@mharding-hpe  

if not apis.activity_exists(activity) i.e if it returs in the scenario where it does not successfully return that activity in existing or backend is not working as excepted 
msg = f"ERROR: Activity {activity} does not exist. or Backend is not working properly"
print_no_of_test_passed(msg, count) . This where we print summary of number of test cases passed and and number of test cases pending and exit in the function with an exit code 1 

def print_no_of_test_passed(ex, count: int):
    print(f"ERROR: {ex}")
    print("~" * 50)
    print(f"INFO: TOTAL test cases passed: {count} test cases skipped: {15 - count}")
    print("~" * 50)
    sys.exit(1)

Reply:
`
def print_no_of_test_passed(ex, count: int):
    print(f"ERROR: {ex}")
    print("~" * 50)
    print(f"INFO: TOTAL test cases passed: {count} test cases skipped: {15 - count}")
    print("~" * 50)
    sys.exit(1)`